### PR TITLE
Update docs to explain use of -g and --files to search for paths.

### LINKED ISSUE #284: [QUESTION/SUGGESTION] Does/should ripgrep support ag/ack's -g flag for searching filenames?
Per the ack documentation:

> -g PATTERN Print files where the relative path + filename matches PATTERN.  This option can be combined with --color to make it easier to spot the match.

I've searched the documentation and can't find this feature. Is this something that you'd be interested in for ripgrep? I'd be happy to submit a PR for it - I'm not a Rust expert but would love to jump in.

Thanks for writing such a great search tool!

### PR DESCRIPTION
**Pull request by YPCrumble**
@BurntSushi this updates docs to resolve #284, #193, #54, #75, #91, #48. Explains the combined use of `-g` and `--files` flags to replicate the `-g` option in ag/ack.

Looks like I used a later version of pandoc to compile the docs that perhaps did things a little differently. It complained about using sed's `-i` flag (`sed: -i may not be used with stdin`).

Please let me know if any issues/suggestions from your end.

Thank you for your patience and thanks again for building ripgrep!

### CODE REVIEW

File: doc/rg.1.md [hunk 1]
```diff
@@ -136,6 +144,10 @@ Project home page: https://github.com/BurntSushi/ripgrep
 --files
 : Print each file that would be searched (but don't search).
 
+    Combine with the -g flag to return matched paths, for example:
+
+    rg -g '<pattern>' --files
```
Comment (BurntSushi) [inline doc/rg.1.md]: As above, change this to `<glob>`.

**Comment by BurntSushi** [comment]
Oh, and could you also put the string `Fixes #284` at the bottom of your commit message? I'm trying to keep things tidy. :-) Thanks!

**Comment by YPCrumble** [comment]
@BurntSushi thank you for your review! Made those changes. Let me know if what I did with the commit message isn't what you meant? Also I escaped the < and > characters around `<glob>` in `rg.1.md` because it appeared that not escaping them made them not appear in `rg.1`. Please me know if this was a mistake?

**Comment by BurntSushi** [comment]
@YPCrumble Looks good to me! I verified that `rg.1` looks right by running `man -l doc/rg.1`.

> Let me know if what I did with the commit message isn't what you meant?

Almost. Ideally, simple PRs like this should just be a single commit. You can do this by amending your previous commit and force pushing your branch, and that commit message is what should contain `Fixes #284` in it. I just went ahead and did that for you though. Thanks again!

**Comment by YPCrumble** [comment]
@BurntSushi got it - thanks for the explanation and making that adjustment for me. Didn't realize I can just take a look at the docs without compiling the code. Glad to have been able to contribute!

**Pull request merged**